test/hotspot/jtreg/serviceability/jvmti/vthread/CarrierThreadEventNotification/libCarrierThreadEventNotification.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <jni.h>
+#include <jvmti.h>
+#include <stdio.h>
+#include <string.h>
+#include "jvmti_common.hpp"
+
+
+extern "C" {
+
+// set by Agent_OnLoad
+static jvmtiEnv* jvmti = nullptr;
+
+static jthread* tested_cthreads = nullptr;
+static jint cthread_cnt = 0;
+
+static const char* CTHREAD_NAME_START = "ForkJoinPool";
+static const int CTHREAD_NAME_START_LEN = (int)strlen("ForkJoinPool");
+
+static jint
+get_cthreads(JNIEnv* jni, jthread** cthreads_p) {
+  jthread* tested_cthreads = NULL;
```

Comment:
This local variable has the same name as global.
I'd suggest to rename the local var or remove it (and the function should set both `tested_cthreads` and ` cthread_cnt`)

Reply:
Thanks. Renamed the local to `cthreads` and the global to `carrier_threads`.